### PR TITLE
HAWQ-1592. vectorized data types initialization and relevant function …

### PR DESCRIPTION
**Pull request by zhangjackey**
…definetion
1.Create vectorized types(vint2, vint4, vint8, vfloat4, vfloat8, vbool).
2.Create the operator functions of the vectorized types.
3.Add the test case of the operators.

### CODE REVIEW

**Comment by wengyanqing** [comment]
Great to have this PR. Would you add some instructions about how to add a new vectorized type and related functions to this framework ?

**Comment by wengyanqing** [comment]
LGTM. +1

File: contrib/vexecutor/vadt.c [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+ * e.g. Datum vint2in(PG_FUNCTION_ARGS)
+ */
+#define FUNCTION_IN(type, typeoid, MACRO_DATUM) \
+PG_FUNCTION_INFO_V1(v##type##in); \
+Datum \
+v##type##in(PG_FUNCTION_ARGS) \
+{ \
+    char *intString = PG_GETARG_CSTRING(0); \
+    v##type *res = NULL; \
+    char tempstr[MAX_VECTOR_SIZE] = {0}; \
```
Comment (interma) [inline contrib/vexecutor/vadt.c]: Use a new DEFINE for the temp string length?

File: contrib/vexecutor/vadt.c [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+PG_FUNCTION_INFO_V1(v##type##out); \
+Datum \
+v##type##out(PG_FUNCTION_ARGS) \
+{ \
+	v##type * arg1 = (v##type *) PG_GETARG_POINTER(0); \
+	int len = arg1->header.dim; \
+    int i = 0; \
+	char *rp; \
+	char *result; \
+	rp = result = (char *) palloc0(len * 16 + 1); \
```
Comment (interma) [inline contrib/vexecutor/vadt.c]: Use a DEFINE for the const value 16?

File: contrib/vexecutor/vadt.c [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+	for (i = 0; i < len; i++) \
+	{ \
+		if (i != 0) \
+			*rp++ = ' '; \
+		strcat(rp, DatumGetCString(DirectFunctionCall1(type##out, MACRO_DATUM(arg1->values[i]))));\
```
Comment (interma) [inline contrib/vexecutor/vadt.c]: strncat() is better.

Comment (zhangjackey) [reply contrib/vexecutor/vadt.c]: the OUT functions may keep it ok, otherwise, we should get the length of the string.

**Comment by interma** [comment]
merged

**Pull request closed without merging**